setup.py

```diff
@@ -14,12 +14,17 @@
 import struct
 import subprocess
 import sys
+import tempfile
 import warnings
 from collections.abc import Iterator
-from typing import Any
+from typing import TYPE_CHECKING, Any
 
 from setuptools import Extension, setup
 from setuptools.command.build_ext import build_ext
+from setuptools.errors import CompileError
+
+if TYPE_CHECKING:
+    import distutils.ccompiler
```

Comment:
Let's avoid some type-checking only imports:
```suggestion

from setuptools import Extension, setup
from setuptools.command.build_ext import build_ext
from setuptools.errors import CompileError

TYPE_CHECKING = False
if TYPE_CHECKING:
    import distutils.ccompiler
    from collections.abc import Iterator
    from typing import Any
```